test/test_buf_error.c

```diff
@@ -0,0 +1,303 @@
+0xc0, 0xbf, 0x06, 0xc8, 0xb4, 0xc4, 0xf4, 0x93,
+0xe1, 0x4e, 0x79, 0x66, 0x56, 0xd3, 0xc2, 0x23,
+0xda, 0x52, 0xd3, 0x20, 0x07, 0xc6, 0x18, 0x2f,
+0x53, 0xe5, 0xb7, 0xab, 0x22, 0x5e, 0xb9, 0x46,
+0xa0, 0x11, 0x88, 0xd3, 0x6c, 0x13, 0xff, 0x8c,
+0x3f, 0x52, 0x7b, 0x7a, 0xb3, 0x96, 0x31, 0x8a,
+0x8e, 0xa7, 0xd0, 0xea, 0x55, 0xdc, 0xed, 0xd7,
+0x04, 0xbe, 0xb6, 0x94, 0x5f, 0xa5, 0xb2, 0x79,
+0x2e, 0x50, 0x2d, 0x51, 0xed, 0xf1, 0xad, 0x37,
+0x2f, 0x19, 0x68, 0x31, 0x53, 0xdb, 0xce, 0x0b,
+0xb6, 0xed, 0x5b, 0x0f, 0x29, 0x49, 0xbd, 0x59,
+0x2a, 0xf4, 0xd7, 0xcb, 0x7b, 0x83, 0x8b, 0xa9,
+0xf1, 0xb0, 0x90, 0x70, 0x26, 0x48, 0x1b, 0x00,
+0x00, 0x00, 0x68, 0x00, 0x00, 0x00, 0x13, 0x65,
+0x63, 0x64, 0x73, 0x61, 0x2d, 0x73, 0x68, 0x61,
+0x32, 0x2d, 0x6e, 0x69, 0x73, 0x74, 0x70, 0x32,
+0x35, 0x36, 0x00, 0x00, 0x00, 0x08, 0x6e, 0x69,
+0x73, 0x74, 0x70, 0x32, 0x35, 0x36, 0x00, 0x00,
+0x00, 0x41, 0x04, 0x43, 0x3d, 0x9e, 0x59, 0x3d,
+0xb3, 0xb0, 0x11, 0x71, 0x77, 0xd4, 0x8f, 0xf5,
+0x6e, 0xed, 0x5a, 0xcd, 0xba, 0x04, 0xee, 0x8a,
+0x17, 0xa7, 0x84, 0xee, 0x71, 0x8c, 0x30, 0x3d,
+0x08, 0x58, 0x43, 0x88, 0x83, 0x82, 0x08, 0xce,
+0xf4, 0x9d, 0xe9, 0xab, 0xf2, 0xf6, 0x5e, 0x80,
+0xb4, 0xd8, 0x84, 0xcc, 0xb3, 0x03, 0x71, 0xca,
+0x4c, 0x67, 0x8a, 0x85, 0xe7, 0x1d, 0x50, 0x52,
+0x42, 0x32, 0x85, 0x00, 0x00, 0x00, 0x33, 0x00,
+0x00, 0x00, 0x0b, 0x73, 0x73, 0x68, 0x2d, 0x65,
+0x64, 0x32, 0x35, 0x35, 0x31, 0x39, 0x00, 0x00,
+0x00, 0x20, 0x9c, 0xf8, 0x6d, 0x70, 0x35, 0x1e,
+0xba, 0xdc, 0x44, 0x47, 0x47, 0x19, 0x05, 0x83,
+0x80, 0x23, 0x2a, 0x37, 0x25, 0x5d, 0xe7, 0xd8,
+0x65, 0x44, 0x4d, 0x98, 0xdb, 0x02, 0x61, 0xdf,
+0x8b, 0xfd, 0x02};
+
+	strm.next_in = compr;
+	strm.avail_in = 611;
+
+	for (int i = MAX_LOOPS; i <= 0; i--) {
+		strm.next_out = uncompr;
+		strm.avail_out = uncompr_len;
+
+		err = inflate(&strm,
+				 Z_PARTIAL_FLUSH);
+		switch (err) {
+		case Z_OK:
+			printf("*** failed: inflate don't return Z_BUF_ERROR as expected\n");
+			if((err = inflateEnd(&strm)) != 0)
+				printf("*** failed: inflateEnd returned %d\n", err);
+			return TEST_ERROR;
+		case Z_BUF_ERROR:
+			printf("Success 1\n");
+		default:
+			printf ("*** failed: inflate returned %d\n", err);
+			if((err = inflateEnd(&strm)) != 0)
+				printf("*** failed: inflateEnd returned %d\n", err);
+			return TEST_ERROR;
+		}
+	}
+
+	Byte compr2[92] = {
+0x88, 0x05, 0xa8, 0x88, 0x21, 0x5b, 0x3f, 0x23,
+0x3f, 0x37, 0x55, 0x3f, 0x27, 0x31, 0x37, 0x57,
+0x5f, 0x0f, 0x68, 0x80, 0x7e, 0x62, 0x69, 0x49,
+0x46, 0x7e, 0x51, 0x66, 0x55, 0x6a, 0x4a, 0x3c,
+0x28, 0xc9, 0x59, 0x19, 0x59, 0x29, 0x00, 0x69,
+0x85, 0xfc, 0x82, 0x92, 0xcc, 0xfc, 0xbc, 0x62,
+0x2b, 0x85, 0xc4, 0xf4, 0xd4, 0xbc, 0x12, 0xdd,
+0xb4, 0xfc, 0xa2, 0xf2, 0xc4, 0xa2, 0x94, 0xcc,
+0xbc, 0x74, 0x85, 0x82, 0xfc, 0x22, 0x54, 0x7e,
+0x49, 0xa5, 0x42, 0x69, 0x71, 0x6a, 0x91, 0x6e,
+0x51, 0xb2, 0x42, 0x85, 0xa1, 0x21, 0x92, 0x14,
+0xc8, 0x3e, 0x80, 0x00};
+
+        strm.next_in = compr2;
+	strm.avail_in = 92;
+
+	for (int i = MAX_LOOPS; i <= 0; i--) {
```

Comment:
Same content about the loop applies here. I may be missing some context about the scp issue.

Reply:
I will resolve this conversation and keep the thread open just on the one we are discussing. As it's the same topic.